### ISSUE TITLE
Toplevel `__init__.py` imports considered invalid if partial stubs are available and do not include it

### ISSUE THREAD

**Issue by zmievsa** [microsoft/pyright#10018]
**Describe the bug**
Previously, we used https://github.com/zmievsa/tortoise-orm-stubs as type stubs for https://github.com/tortoise/tortoise-orm and everything worked well. However, in versions 1.1.395 and 1.1.396 we started seeing a new bug: toplevel library `__init__.py` is not defined in stubs so its definitions were always taken from tortoise-orm directly but now direct imports from that file show `unknown import symbol` errors.

**Code or Screenshots**
First, install tortoise-orm==0.21.7 and tortoise-orm-stubs==1.0.2

```python
from tortoise import Tortoise
```

You will get the following error only on latest versions:
```
error: "Tortoise" is unknown import symbol (reportAttributeAccessIssue)
```

**VS Code extension or command-line**
Command-line: 1.1.396. Vscode doesn't have this problem.
python 3.12, macos and ubuntu

**Comment by erictraut** [comment]
I'm not able to repro the problem with the steps provided. I've installed the two specified versions of these libraries (both the core library and the stub library) and pasted the code into a new source file, but I don't see any type checking error. 

Can you confirm that you're able to repro with the above steps with a clean project and venv? Is it possible that there are additional packages installed in your environment that could affect import resolution? Or custom pyright settings?

What happens if you right-click on `Tortoise` in the above code sample and choose `Go To Declaration`? For me, it takes me to the `site-packages/tortoise/__init__.py` file, which shows that import resolution is working correctly.

If I then type `from tortoise.fields import BigIntField` and right-click on `BigIntField` and choose `Go To Declaration`, it takes me to the `site-packages/tortoise-stubs/fields/__init__.pyi` file, which is what I'd expect.

**Comment by zmievsa** [comment]
```bash
❯ uv --version
uv 0.5.26 (Homebrew 2025-01-30)
❯ uv init --python=3.12
Initialized project `testiks`
❯ uv add pyright tortoise-orm tortoise-orm-stubs
Using CPython 3.12.9 interpreter at: /opt/homebrew/opt/python@3.12/bin/python3.12
Creating virtual environment at: .venv
Resolved 10 packages in 141ms
Prepared 2 packages in 63ms
Installed 9 packages in 170ms
 + aiosqlite==0.21.0
 + iso8601==2.1.0
 + nodeenv==1.9.1
 + pypika-tortoise==0.5.0
 + pyright==1.1.396
 + pytz==2025.1
 + tortoise-orm==0.24.2
 + tortoise-orm-stubs==1.0.2
 + typing-extensions==4.12.2
❯ echo "from tortoise import Tortoise" > hello.py
❯ uv run pyright .
/Users/zmievsa/code/trash/testiks/hello.py
  /Users/zmievsa/code/trash/testiks/hello.py:1:22 - error: "Tortoise" is unknown import symbol (reportAttributeAccessIssue)
1 error, 0 warnings, 0 informations
```

**Comment by erictraut** [comment]
Thanks, I'm able to repro the problem using your repro steps. Strangely, it doesn't repro with the VS Code extension, only with the command-line.

I've narrowed down the cause of the regression to [this bulk PR from the pylance team](https://github.com/microsoft/pyright/commit/2714cdb466e22cad062e175e07081876992ab5be). I don't have visibility into the individual PRs that comprise this commit, so the pylance team is in a better position than me to further analyze and debug the issue. It appears that there were some changes in this commit relating to partial type stubs, so I'm guessing that's the cause.

@rchiodo, could you please look at this or assign it to someone else on the pylance team as appropriate? Thanks!

**Comment by rchiodo** [comment]
I think I know what the problem is. We separated out the partial stub support into a separate service but didn't create the service in the pyright case.

**Comment by zmievsa** [comment]
Thanks, guys!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.397.
